### PR TITLE
Weppy ember 0.0.1 wip

### PR DESCRIPTION
**Pull request by kenkouot**
@rogatty
Still a bit of work to be done but this refactor from Igor's original work clarifies the workflow a bit. In order to use `weppy-ember`, the developer must initialize it in the appropriate place, in Ember's `App.initializer` method. Here, the developer can pass in a config object with specific implementations for logging and persistence methods.

``` ts
App.initializer({
  initialize: () => {
    WeppyEmber.initialize({ /** options **/});
  }
});
```

### CODE REVIEW

File: src/weppy-ember.ts [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+	}
+
+	export interface WeppyEmberConfig {
+		enableAjaxFilter: boolean;
+		enableLogging: boolean;
+		error?: (message: string) => void;
+		log?: (message: string) => void;
+		minDuration: number;
+		send: (events: any[], metrics: MetricsReport) => void;
+		warn?: (message: string ) => void;
```
Comment (mklucsarits) [inline src/weppy-ember.ts]: Extra space inside parentheses

File: src/weppy-ember.ts [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+		minDuration: 50,
+		log: (message) => { console.log(message); },
+		send: (events, metrics) => { console.log('Sending (Dry Run)\n============================\n', metrics); },
+		warn: (message) => { console.log('warn: ' + message); },
+		error: (message) => { console.log('error: ' + message); }
+	};
+
+	var traceStack = [];
+
+	function getLastTraceItem () {
```
Comment (mklucsarits) [inline src/weppy-ember.ts]: I don't recall offhand our standard spacing for functions, but the spacing between function names and parentheses is inconsistent for these functions.

File: src/weppy-ember.ts [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+		return traceStack[traceStack.length - 1];
+	}
+
+	function getConfigMethod(method, ...args) {
+		if (config.enableLogging && typeof config[method] === 'function') {
+			return config[method].apply(this, args);
+		}
+		console.warn('Method ' + method + 'not found');
```
Comment (mklucsarits) [inline src/weppy-ember.ts]: Need a space before "not found"

File: src/weppy-ember.ts [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+				ret = decorator.apply(this, arguments);
+			} catch (e) {
+				throw e;
+			}
+			return (typeof orig === 'function' ? orig : () => {}).apply(this, arguments);
+		}
+	}
+
+	function aliasMethodChain ($, method, label, wrapper) {
+		var e = $[method];
```
Comment (mklucsarits) [inline src/weppy-ember.ts]: Perhaps replace single-letter vars here with something more descriptive.

File: src/weppy-ember.ts [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+			}
+		}
+
+		resume () {
+			this.pause();
+			traceStack.push(this);
+		}
+
+		finalize () {
+			if (this.finalized === !0) {
```
Comment (mklucsarits) [inline src/weppy-ember.ts]: Why !0 ?

**Comment by rogatty** [comment]
Now I wonder if this library should be called weppy-ember. It doesn't actually have anything to do with Weppy, right? Initially I thought about tight coupling it with Weppy but now I see that it will come in a form of configuration. Maybe we should rename it while it's still easy. It's more like ember-performance-monitor or whatever ¯_(ツ)_/¯

**Comment by kenkouot** [comment]
@rogatty That's actually a really good point. Let's brainstorm in channel. I've made some more changes in the meantime.

**Pull request merged**